### PR TITLE
libxo 0.8.4 (new formula)

### PR DESCRIPTION
**Pull request by trombonehero**
- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/libxo.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  option "without-test", "Skip build-time testing (not recommended)"
+
+  depends_on "autoconf" => :build
+  depends_on "libtool" => :build
+
+  def install
+    args = %W[--prefix=#{prefix}]
+
+    system "./configure", *args
+    system "make", "test" if build.with? "test"
```
Comment (ilovezfs) [inline Formula/libxo.rb]: we don't allow this except for security software like openssl

File: Formula/libxo.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
+class Libxo < Formula
+  desc "Allows an application to generate text, XML, JSON, and HTML output"
+  homepage "https://juniper.github.io/libxo/libxo-manual.html"
+  url "https://github.com/Juniper/libxo/releases/download/0.8.4/libxo-0.8.4.tar.gz"
+  sha256 "3b3a33b706075fdc478782fdc65db3e7f9167550f7d71b95e704236a797e7fd3"
+  head "https://github.com/Juniper/libxo"
```
[the remark this answers is not in the document]
Comment (ilovezfs) [reply Formula/libxo.rb]: and actually let's just remove head

File: Formula/libxo.rb [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+  option "without-test", "Skip build-time testing (not recommended)"
+
+  depends_on "autoconf" => :build
```
Comment (ilovezfs) [inline Formula/libxo.rb]: shouldn't be needed since we're using a release tarball

File: Formula/libxo.rb [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  depends_on "libtool" => :build
```
Comment (ilovezfs) [inline Formula/libxo.rb]: ditto

File: Formula/libxo.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (DomT4) [reply Formula/libxo.rb]: > needs .git on the end

Another one for the _"Why is this not audited"_ pile.

[the remark this answers is not in the document]
Comment (ilovezfs) [reply Formula/libxo.rb]: you yelled at me when I suggested that once I believe lol

[the remark this answers is not in the document]
Comment (DomT4) [reply Formula/libxo.rb]: Did I? 🙈. It's possible you bothered me with something else & I ended up snapping at you over a relatively minor thing instead. That happens 🙈.

[the remark this answers is not in the document]
Comment (ilovezfs) [reply Formula/libxo.rb]: It would be better if I didn't remember things like that.

[the remark this answers is not in the document]
Comment (DomT4) [reply Formula/libxo.rb]: I'm quite prone to letting annoyance build up within me until something pokes it across the line, and then I get really annoyed about that something, however small. Sometimes it's worth asking me if I'm actually annoyed with what's being discussed or 😅. But yeah, if you did suggest this at some point, I've come around to the idea.

File: Formula/libxo.rb (hunk 1 above)
Comment (trombonehero) [reply Formula/libxo.rb]: Sorry, just to be clear: you don't allow running tests or skipping tests?

File: Formula/libxo.rb (hunk 4 above)
Comment (trombonehero) [reply Formula/libxo.rb]: Without the `libtool` dependency I get the following:

```
Making install in libxo                                                         
  CC       libxo.lo                                                             
  CC       xo_encoder.lo                                                        
/bin/sh: glibtool: command not found                                            
/bin/sh: glibtool: command not found
```

You're quite right about `autoconf`, however.

File: Formula/libxo.rb (hunk 1 above)
Comment (ilovezfs) [reply Formula/libxo.rb]: right

Comment (ilovezfs) [reply Formula/libxo.rb]: we don't allow `make check` or `make test`

**Comment by ilovezfs** [comment]
I think this is the diff we need
```
diff --git a/Formula/libxo.rb b/Formula/libxo.rb
index 243138e..2208c74 100644
--- a/Formula/libxo.rb
+++ b/Formula/libxo.rb
@@ -4,15 +4,13 @@ class Libxo < Formula
   url "https://github.com/Juniper/libxo/releases/download/0.8.4/libxo-0.8.4.tar.gz"
   sha256 "3b3a33b706075fdc478782fdc65db3e7f9167550f7d71b95e704236a797e7fd3"
 
-  option "without-test", "Skip build-time testing (not recommended)"
-
   depends_on "libtool" => :build
 
   def install
-    args = %W[--prefix=#{prefix}]
-
-    system "./configure", *args
-    system "make", "test" if build.with? "test"
+    system "./configure", "--disable-debug",
+                          "--disable-dependency-tracking",
+                          "--disable-silent-rules",
+                          "--prefix=#{prefix}"
     system "make", "install"
   end
 ```

File: Formula/libxo.rb [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  def install
+    system "./configure", "--disable-debug",
+                          "--disable-dependency-tracking",
+                          "--disable-silent-rules",
+                          "--prefix=#{prefix}"
+    system "make", "install"
+  end
+
+  test do
+    # Quick sanity check:
```
Comment (ilovezfs) [inline Formula/libxo.rb]: comment is unneeded

**Review by ilovezfs — APPROVED** [review]
Can you please squash this down to one commit? Thanks!

**Comment by trombonehero** [comment]
Squashed as requested.

**Comment by ilovezfs** [comment]
Thanks for your first contribution to Homebrew, @trombonehero! Without people like you submitting PRs we couldn't run this project. You rock!

**Pull request closed without merging**